### PR TITLE
Shrink bcf test files

### PR DESCRIPTION
**Pull request by sem-geologist**
This PR is to improve situation with #2048
### Description of the change
Shrinks the bcf test files by replacing some largest with minimal examples.
The python test code for bcf were adapted too.

### Accidental bug findings and fixes
replaced the manual basename parsing in `bruker.py` to `os.path.basename`,
so now the `metadata.General.original_filename` is identical under both windows and posix.

### Progress of the PR
- [x] Create new smaller replacement bcf test files,
- [x] update test_bruker.py and helper files (numpy, json..),
- [x] replaced manual filename basename parsing to `os.path.basename`
- [x] ready for review.

### CODE REVIEW

**Comment by sem-geologist** [comment]
There is still room for improvement. The biggest offender (>2.5MB) now is version2 bcf file in bcf testing dataset (`bcf_v2_50x50px.bcf`).Unfortunately, Our lab only plans to upgrade to newer Esprit, but I have no idea when this will happen. The present file was donated, but shame on me, I had not kept track by who. If anyone have access to Esprit 2.x could someone produce the most minimal example (image size needs to be set to 100x100 and ROI for hypermap 16x16).

File: hyperspy/tests/io/test_bruker.py [hunk 1]
```diff
@@ -83,56 +80,70 @@ def test_hyperspy_wrap():
...
+            'binned': True,
+            'quantity': 'X-rays (Counts)',
+            'signal_type': 'EDS_SEM'},
+        '_HyperSpy': {
+            'Folding': {'original_axes_manager': None,
+            'original_shape': None,
+            'signal_unfolded': False,
+            'unfolded': False}}}
+
+    #md_ref['General']['original_filename'] = hype.metadata.General.original_filename #NANI???
```
Comment (sem-geologist) [inline hyperspy/tests/io/test_bruker.py]: Looks this was used to hack around the hyperspy metadata original_filename full path leak on windows.

**Comment by sem-geologist** [comment]
Raised issues due to this PR:
* files are too small, compressed file have not enough zlib'ed chunks. Due to that, part of code, which deals with compression chunks, escapes tests (coverage decrease).
~* appveyor fails, I think I found some new bug:
excerpt from appveyor log:~

```
actual = {'date': '2018-09-12', 'original_filename': '16x16_instructively_packed_16bit_compressed.bcf', 'time': '13:01:53', 'title': 'EDX'}
expected = {'date': '2018-09-12', 'original_filename': 'C:\\projects\\hyperspy\\hyperspy\\tests\\io\\bruker_data\\16x16_instructively_packed_16bit_compressed.bcf', 'time': '13:01:53', 'title': 'EDX'}
```

~why on windows hyperspy puts to `metadata.General.original_filename` the full path? on *nx it does not. That is inconsistent. I see previouslyin `test_bruker.py` this was hacked around by this~ 
```python
md_ref['General']['original_filename'] = hype.metadata.General.original_filename
```
~While doing this PR I was wondering why that line is there... after commenting it out I got into appveyor failure.
@francisco-dlp , can something be done that metadata on windows would be identical for that on posix systems?~

**update**: issue solved by replacing manual filename splitting to `os.path.basename` in `bruker.py`

**Comment by ericpre** [comment]
Is it ready for review?

**Comment by sem-geologist** [comment]
@ericpre , unfortunately not. I need to take new bcf sample, the compressed one. Which would be huge enough to have at least 2 internally zlib'ed chunks, but small enough for this PR to make sense. Our SEM lab is quite busy, and I have some other assignments (I am primarily working with electron microprobes (WDS and stuff primarily), the SEM had become only secondary tool for me, and I have no more unlimited access to it). I will let know when this will be ready for review. (P.S. the check list at the top is for a reason).

**Comment by sem-geologist** [comment]
@ericpre , It is ready for review.
The effective size (after compression – the only which actually maters) was reduced from 8.7MB to 3.3MB. I had asked in gitter for someone to try redoing the v2 bcf, but I had not got any bcfs. `bcf_v2_50x50.bcf` at the moment is the largest and is taking 1.6MB (effective size). So probably in future there is some space for more improvement (our lab plans to upgrade to v2, but it is taking very slow).

**Pull request merged**